### PR TITLE
Implements expected behaviour while adding user in test util

### LINKED ISSUE #659: Fix user fixtures and test setup
**Problem**
In real life scenario when a user is first added in the DB they do not have any roles.

But the [fixtures](https://github.com/Real-Dev-Squad/website-backend/blob/develop/test/fixtures/user/user.js) being used to add a new user (during the tests setup) has the roles property pre-populated and we end up with a scenario where a new user added in the DB already has roles. 

Hence we need to fix the fixtures and modify the test setup part where the new user added in DB must not have any roles. 
If we want to add roles it must be done in the same way as it'll be done in a real life scenario

Basically the tests should mimic the business flow and use cases 

**How we found this issue?**
Read this discussion : https://github.com/Real-Dev-Squad/website-backend/pull/654#discussion_r939069800

### PR DESCRIPTION
**Pull request by samarpan1738**
## This PR can only be merged after #649 

- #649 

* Add user in firestore without roles

* Update user's roles after adding the user in firestore

Closes #659 

Please follow the merge order specified here : https://github.com/Real-Dev-Squad/website-backend/issues/672

### CODE REVIEW

File: test/utils/addUser.js [hunk 1]
```diff
@@ -12,7 +12,19 @@ module.exports = async (user) => {
   const isValid = user && Object.keys(user).length !== 0 && user.constructor === Object;
   // Use the user data sent as arguments, else use data from fixtures
   user = isValid ? user : userData[0];
+  /*
+    Making a shallow copy of user allows us to safely modify the user object
+    without affecting the original fixture
+  */
+  user = { ...user };
```
Comment (ankushdharkar) [inline test/utils/addUser.js:19]: Can we use a new variable here please? It's bad practice to reuse param from function params list.

File: test/utils/addUser.js [hunk 2]
```diff
@@ -12,7 +12,19 @@ module.exports = async (user) => {
...
+  const roles = user.roles;
```
Comment (ankushdharkar) [inline test/utils/addUser.js:20]: Can this be named better?

File: test/utils/addUser.js [hunk 3]
```diff
@@ -12,7 +12,19 @@ module.exports = async (user) => {
...
+  delete user.roles;
```
Comment (ankushdharkar) [inline test/utils/addUser.js:21]: Please add a comment here of why we are doing this.

File: test/utils/addUser.js [hunk 4]
```diff
@@ -12,7 +12,19 @@ module.exports = async (user) => {
...
   const { userId } = await users.addOrUpdate(user);
-
+  if (roles) {
+    const persistedUser = (await users.fetchUser({ userId })).user;
+    const currentRoles = persistedUser.roles;
+    const updatedRoles = currentRoles ? { ...currentRoles, ...roles } : roles;
```
Comment (ankushdharkar) [inline test/utils/addUser.js:26]: Will this work if `currentRoles` is an empty object?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: test/utils/addUser.js (hunk 2 above)
Comment (samarpan1738) [reply test/utils/addUser.js]: Surely. Changes done

File: test/utils/addUser.js (hunk 3 above)
Comment (samarpan1738) [reply test/utils/addUser.js]: Done

File: test/utils/addUser.js (hunk 4 above)
Comment (samarpan1738) [reply test/utils/addUser.js]: Yes. It'll work. Removed the ternary

File: test/utils/addUser.js (hunk 1 above)
Comment (samarpan1738) [reply test/utils/addUser.js]: Done.

**Review by ankushdharkar — APPROVED** [review]

**Pull request merged**